### ISSUE TITLE
ModuleNotFoundError: When running RFSwarm-Agent on Python 3.12

### ISSUE THREAD

**Issue by damies13** [damies13/rfswarm#177]
ModuleNotFoundError: When running RFSwarm-Agent on Python 3.12

Occurs on Windows, MacOS & Ubuntu github actions, ~~but interestingly though the help and version screens fail it appears the agent will actually run robots with this error.~~

-----
Traceback (most recent call last):
  File "/Users/runner/work/rfswarm/rfswarm/rfswarm_agent/rfswarm_agent.py", line 34, in <module>
    import pkg_resources
ModuleNotFoundError: No module named 'pkg_resources'

-----

**Comment by damies13** [comment]
https://stackoverflow.com/questions/49923671/are-there-any-function-replacement-for-pip-get-installed-distributions-in-pip

This works on Python 3.9.10
```
import importlib.metadata
>>> dists = importlib.metadata.distributions()
>>> for dist in dists:
...     if "robot" in dist.metadata["Name"]:
...             print(dist.metadata["Name"], dist.version)
... 
robotframework-sshlibrary 3.8.0
robotframework-browser 11.1.1
robotframework-seleniumlibrary 6.0.0
robotframework-jsonvalidator 2.0.0
robotframework 6.0.2
robotframework-imagehorizonlibrary 1.0
robotframework-listenerlibrary 1.0.3
robotframework-SikuliLibrary 2.0.0
robotframework-pythonlibcore 3.0.0
robotframework-faker 5.0.0
robotframework-datadriver 1.7.0
robotframework-perfmon 0.1
robotframework-requests 0.9.4
robotframework-assertion-engine 0.2.0
>>> 

```

**Comment by damies13** [comment]
https://docs.python.org/3.11/library/importlib.metadata.html#module-importlib.metadata

module importlib.metadata
> New in version 3.8.
Changed in version 3.10: importlib.metadata is no longer provisional.

**Comment by damies13** [comment]
Now the agent appears fixed on 3.12, the manager is getting errors when trying to run the test case

Turning this issue into a general any python 3.12 related issues until test case I created for Issue #171 can run sucessfully under python 3.12

New Error (repeats over and over:
```
-----stderr_manager.txt-----
----------------------------------------
Exception occurred during processing of request from ('127.0.0.1', 42518)
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.12.2/x64/lib/python3.12/socketserver.py", line 318, in _handle_request_noblock
    self.process_request(request, client_address)
  File "/opt/hostedtoolcache/Python/3.12.2/x64/lib/python3.12/socketserver.py", line 706, in process_request
    t.start()
  File "/opt/hostedtoolcache/Python/3.12.2/x64/lib/python3.12/threading.py", line 9[92](https://github.com/damies13/rfswarm/actions/runs/7943062028/job/21687004766#step:5:93), in start
    _start_new_thread(self._bootstrap, ())
RuntimeError: can't create new thread at interpreter shutdown
----------------------------------------
```

**Comment by damies13** [comment]
Maybe related

https://github.com/getsentry/sentry-python/issues/2299

**Comment by damies13** [comment]
This looks related

https://stackoverflow.com/questions/77238646/python-http-server-threadinghttpserver-fails-since-update-to-python-3-12

Still need to investigate more

**Comment by damies13** [comment]
Also probbaly related

https://stackoverflow.com/questions/2846653/how-do-i-use-threading-in-python

This is begining to look like a bigger issue than I expected

**Comment by damies13** [comment]
I think I'm being impacted by this bug:
https://github.com/python/cpython/issues/115533

**Comment by damies13** [comment]
This is interesting, looks like the is a workaround with sleep?
https://github.com/python/cpython/issues/113964

**Comment by damies13** [comment]
Fixes for Issue #177 merged into release branch v1.2.1

**Comment by damies13** [comment]
All tests and linter passed
